doc/api.md

```diff
@@ -1,230 +1,692 @@
 | ---- | ---- | ---------------- |
-| contracts   | array | an array of contract information. The attributes of each contract are defined in the following rows. |
-| contract_address | string |  the contract address on the blockchain. |
-| name | string |  the name of the contract.  |
-| archived | boolean | not used.   |
-| current_agreement_id | string | the id of agreement between the device and the data consumer that is currently in effect. |
-| previous_agreements | array | an array of the previous agreement ids that are canceled.  |
-| configure_nonce | string | a unique identifier that is used for contract negotiation. It is set when the negotiation starts and unset when the device starts downloading the workloads. |
-| agreement_creation_time | int64 | the time when the agreement proposal is received from the data consumer and the negotiation process starts. |
-| agreement_execution_start_time | int64 | the time when the device starts the workloads. |
-| agreement_accepted_time | int64 | the time when the agreement is finalized and the device will start receiving payment tokens. |
-| private_environment_additions | json | the private environment variables that will be passed into the workload containers. |
-| environment_additions | json | the public environment variables that will be passed into the workload containers. They will be written into the contract agreement. |
-| current_deployment | json | contains the information of the workloads. The key is the name of the workload and the value is the result of [/containers/<id> docker remote API call](https://docs.docker.com/engine/reference/api/docker_remote_api_v1.24/#/inspect-a-container) for the  workload container. Please refer to the link for details. |
+| services   | json | a list of services with each service containing policy and attributes. |
+| services.{name}.policy | json | the policy for the service. This policy will be used to negotiate an agreement with the agbot. |
+| services.{name}.attributes | array | an array of the user defined service attributes for the service. The attributes will be passed into the workloads environmental variables after the agreement is reached. |
+
+policy
+
+| name | type | description |
+| ---- | ---- | ---------------- |
+| header | json|  the header of the policy. It includes the name and the version of the policy. |
+| apiSpec | array | an array of api specifications. Each one includes a URL pointing to the definition of the API spec, the version of the API spec in OSGI version format, whether or not exclusive access to this API spec is required and the hardware architecture of the API spec implementation. | 
+| agreementProtocols | array | an array of agreement protocols. Each one includes the name of the agreement protocol.|
+| valueExchange | json | It includes the type of value exchange, the value being exchanged, the number of seconds between payments and a token used to identify the user of the value. |
```

Comment:
I would prefer to not document several of these fields until they actually mean something. Please omit valueExchange, resourceLimits, dataVerification, and proposalRejection. It might be good to also put in a statement like this "The policy also contains other fields that are unused and therefore not documented."